backend/src/test/java/gov/cdc/usds/simplereport/utils/BulkUploadResultsToFhirTest.java

```diff
@@ -92,9 +95,77 @@ void convertExistingCsv_success() {
     assertThat(resourceUrls).hasSize(14);
   }
 
+  @Test
+  void allFieldsCsv_TestOrderedCodeMapped() throws IOException {
+    byte[] input = loadCsv("testResultUpload/test-results-upload-all-fields.csv").readAllBytes();
+
+    var serializedBundles =
+        sut.convertToFhirBundles(new ByteArrayInputStream(input), UUID.randomUUID());
+    var mappingIterator = getIteratorForCsv(new ByteArrayInputStream(input));
+
+    int index = 0;
+    while (mappingIterator.hasNext()) {
+      var csvRow = mappingIterator.next();
+      var inputOrderedCode = csvRow.get("test_ordered_code");
+      var inputPerformedCode = csvRow.get("test_performed_code");
+
+      var bundle = serializedBundles.get(index++);
+      var deserializedBundle = (Bundle) parser.parseResource(bundle);
+
+      var serviceRequestEntry =
+          deserializedBundle.getEntry().stream()
+              .filter(entry -> entry.getFullUrl().contains("ServiceRequest/"))
+              .findFirst()
+              .orElseThrow(
+                  () -> new AssertionError("Expected to find ServiceRequest, but not found"));
+      var serviceRequest = (ServiceRequest) serviceRequestEntry.getResource();
+
+      var mappedCode = serviceRequest.getCode().getCoding().stream().findFirst().get().getCode();
+
+      // value is mapped
+      assertThat(mappedCode).isEqualTo(inputOrderedCode);
+      // value is not defaulted to performed code
+      assertThat(inputOrderedCode).isNotEqualTo(inputPerformedCode);
+    }
+  }
+
+  @Test
+  void validCsv_TestOrderedCodeDefaultedToPerformedCode() throws IOException {
+    byte[] input = loadCsv("testResultUpload/test-results-upload-valid.csv").readAllBytes();
+
+    var serializedBundles =
+        sut.convertToFhirBundles(new ByteArrayInputStream(input), UUID.randomUUID());
+    var mappingIterator = getIteratorForCsv(new ByteArrayInputStream(input));
+
+    int index = 0;
+    while (mappingIterator.hasNext()) {
+      var csvRow = mappingIterator.next();
+      var inputOrderedCode = csvRow.get("test_ordered_code");
+      var inputPerformedCode = csvRow.get("test_performed_code");
+
+      var bundle = serializedBundles.get(index++);
```

Comment:
Should this test use a csv that has multiple rows as well?

Reply:
I don't think it needs to, but I left the iterating piece in place because the file _could_ have multiple lines and the test should still be valid